project/Build.scala

```diff
@@ -207,11 +209,15 @@ object DottyBuild extends Build {
         "org.scala-sbt" % "api" % sbtVersion.value % "test",
         "org.specs2" %% "specs2" % "2.3.11" % "test"
       ),
-      version := "0.1.1-SNAPSHOT",
-      // The sources should be published with crossPaths := false, the binaries
-      // are unused so it doesn't matter.
+      version :=
+        "0.1.1-" + VersionUtil.commitDate + "-" + VersionUtil.gitHash + "-SNAPSHOT",
```

Comment:
I'm not sure if it is a good idea to publish hundreds of distinct snapshots on maven and ask users to specify git-hash by hand. I'd prefer to publish one snapshot and have version information reported by REPL. Something similar to https://github.com/hmrc/sbt-git-stamp may be in handy for implementing this.